opal/corelib/string.rb

```diff
@@ -436,6 +436,7 @@ def gsub(pattern, replacement = undefined, &block)
       }
 
       var result = '', match_data = nil, index = 0, match, _replacement;
+      var onlyAnchor = /\/[\^\$]\//.test(pattern.toString())
```

Comment:
happy to rewrite it - though I'm not sure how else youd like to do it. Perhaps checking `pattern.source` instead of stringifying (thats all that browser engines are doing when tostring is being called, really)